packages/runtime/src/runtime.ts

```diff
@@ -346,18 +344,23 @@ export class Runtime {
     }
   }
 
-  // Delete application from account's state and global state
+  /**
+   * Delete application from account's state and global state
+   * @param appId Application Index
+   */
   deleteApp (appId: number): void {
     if (!this.store.globalApps.has(appId)) {
       throw new TealError(ERRORS.TEAL.APP_NOT_FOUND, { appId: appId });
     }
-    const accountAddr = this.store.globalApps.get(appId)?.creator;
+    const accountAddr = this.store.globalApps.get(appId);
     if (accountAddr === undefined) {
       throw new TealError(ERRORS.TEAL.ACCOUNT_DOES_NOT_EXIST);
     }
     const account = this.assertAccountDefined(this.store.accounts.get(accountAddr));
+
     account.deleteApp(appId);
     this.store.globalApps.delete(appId);
+    this.store.accounts.set(account.address, account);
```

Comment:
is `this.assertAccountDefined(this.store.accounts.get(accountAddr))` returning a deepCopy ?

Reply:
yes, similar to addApp, did you figure what was going on with addAPp ?

Reply:
When i tested `addApp` it wasn't returning a deepCopy. That's why i asked here

Reply:
verified it, working after removing it, will have a look at each function for this too.